src/main/java/net/minestom/server/command/builder/arguments/Argument.java

```diff
@@ -220,6 +217,18 @@ public boolean hasSuggestion() {
         return suggestionCallback != null;
     }
 
+    /**
+     * Maps this argument's output to another result.
+     *
+     * @param mapper The mapper to use (this argument's input --> desired output)
+     * @param <O> The type of output expected.
+     *
+     * @return A new ArgumentMap that can get this complex object type.
+     */
+    public <O> ArgumentMap<T, O> map(ArgumentMapper<T, O> mapper) {
```

Comment:
Could you annotate it with `@Beta`?

I am ready to merge it, but we might find an alternate path in the future